### PR TITLE
AT-5885 List all ATAT users

### PR DESCRIPTION
**Pull request by jGonzalez-CCPO**
Description
As a CCPO admin, I want to be able to view all users, so that I can contact users about their portfolio, troubleshoot access issues, or modify ATAT permissions
 
See below for an image of the component, and see design specs on the Users List View Abstract page.
Note: Abstract/Invision links will only be available until June 2021. A PNG version of the view from the above Abstract link is retained for reference on Confluence here: Users List View.png. Please see Design Files and Assets  in Confluence for instructions on how you can view design specs using Sketch.

Acceptance Criteria
Columns for the Users List View include the below for each user:
Name
Organization
DoD ID
Portfolio(s)
Last login date
The below columns should be sortable ascending and descending:
Name - alphabetically
Organization - alphabetically
Last login - chronologically
The default sort is name alphabetically

Additional request for adding ASC/DESC to the filter

### CODE REVIEW

File: tests/routes/admin/test_admin.py [hunk 1]
```diff
@@ -48,3 +51,14 @@ def test_user_page(user_session, client):
...
+
+
+def test_all_users_page(user_session, client):
+    ccpo = UserFactory.create_ccpo()
+    user2 = UserFactory.create()
+    user_session(ccpo)
+    response = client.get(
+        str(url_for("admin.all_users_page")) + "?order-by=last_name&order-asc=-1"
+    )
+    assert response.status_code == 200, "ccpo user should access all users page"
```
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply tests/routes/admin/test_admin.py:63]: We also probably want to add a test to ensure that non-CCPO users cannot access this page.

File: atat/routes/admin/admin.py [hunk 2]
```diff
@@ -59,22 +60,41 @@ def dashboard():
...
+        {
+            "id": str(user.id),
+            "full_name": user.full_name,
+            "branch": service_branch_label(user.service_branch),
+            "dod_id": user.dod_id,
+            "portfolios_descriptions": get_portfolios_descriptions_from_user(user),
+            "last_login": user.last_login.strftime("%b %d, %Y \n %H:%M %Z")
+            if user.last_login is not None
+            else "",
+            "last_login_unix": int(user.last_login.timestamp())
```
Comment (kylelaker-ccpo) [inline atat/routes/admin/admin.py:76]: We don't use this field and it's duplicative of `last_login` but less well-formatted. Why is it here?

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]
Please take a look at this feedback and provide responses or corrections.

**Review by kylelaker-ccpo — CHANGES_REQUESTED** [review]

File: atat/routes/admin/admin.py [hunk 3]
```diff
@@ -59,22 +73,37 @@ def dashboard():
...
-            service_branch_label(user.service_branch),
-        )
+        {
+            "id": str(user.id),
+            "full_name": user.full_name,
+            "branch": service_branch_label(user.service_branch),
+            "dod_id": user.dod_id,
+            "portfolios_descriptions": get_portfolios_descriptions_from_user(user),
+            "last_login": get_date_and_timestamp(user.last_login)[0],
+            "last_login_unix": get_date_and_timestamp(user.last_login)[1],
```
[the remark this answers is not in the document]
Comment (jGonzalez-CCPO) [reply atat/routes/admin/admin.py:87]: can I do that in the array loop?

[the remark this answers is not in the document]
Comment (jGonzalez-CCPO) [reply atat/routes/admin/admin.py:87]: jum, i think i get i idea

File: tests/routes/admin/test_admin.py [hunk 4]
```diff
@@ -48,3 +54,25 @@ def test_user_page(user_session, client):
...
+def test_all_users_page(user_session, client):
+    ccpo = UserFactory.create_ccpo()
+    user2 = UserFactory.create()
+    user_session(ccpo)
+    response = client.get(
+        str(url_for("admin.all_users_page")) + "?order-by=last_name&order-asc=-1"
+    )
+    assert (
+        response.status_code == 200
+    ), "Accessing the users page should work for CCPO users"
```
[the remark this answers is not in the document]
Comment (jGonzalez-CCPO) [reply tests/routes/admin/test_admin.py:68]: true working on it

File: atat/routes/admin/admin.py [hunk 5]
```diff
@@ -12,7 +14,7 @@
 bp.context_processor(atat_context_processor)
 
 
-def get_portfolios_from_user(user: User = None):
+def get_portfolios_descriptions_from_user(user: User = None):
```
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply atat/routes/admin/admin.py]:
```
from typing import Optional
```

[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply atat/routes/admin/admin.py]: It's in the Python standard library.

File: atat/routes/admin/admin.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kylelaker-ccpo) [reply atat/routes/admin/admin.py]: Even better!

**Pull request merged**